packages/gatsby/src/query/queue.js

```diff
@@ -100,15 +92,22 @@ const processBatch = async (queue, jobs, activity) => {
     return Promise.resolve()
   }
 
-  const runningJobs = jobs.map(job =>
-    pushJob(queue, job).then(v => {
-      if (activity.tick) {
-        activity.tick()
-      }
-      return v
-    })
-  )
-  return Promise.all(runningJobs)
+  return new Promise((resolve, reject) => {
+    if (activity.tick) {
+      queue.on(`task_finish`, () => activity.tick())
+    }
+
+    queue
+      .on(`task_failed`, (...err) => {
+        // Note: the first arg is the path, the second the error
+        reject(err)
+        return err
```

Comment:
Curious, why would we `return` from the listener?

Reply:
[Looks like](https://github.com/diamondio/better-queue/blob/master/lib/worker.js#L15) it's using https://nodejs.org/api/events.html and in there I don't see a use for the return value. So I'll drop it :) Good suggestion